### PR TITLE
init: Extracts Availability Zone from ECS Task Metadata.

### PR DESCRIPTION
**Pull request by johnjameswhitman**
*Issue #, if available:* #538

*Description of changes:* Makes a task's AZ available as an environment variable.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: init/fluent_bit_init_process_test.go [hunk 1]
```diff
@@ -94,18 +97,20 @@ func TestSetECSTaskMetadata(t *testing.T) {
...
+		ECS_TASK_ARN:          "arn:aws:ecs:us-west-2:111:task/ecs-local-cluster/37e8",
+		ECS_TASK_ID:           "56461",
+		ECS_FAMILY:            "esc-task-definition",
+		ECS_LAUNCH_TYPE:       "EC2",
+		ECS_REVISION:          "1",
+		ECS_TASK_DEFINITION:   "esc-task-definition:1",
 	}
 
-	expectedContent1 := "export FLB_AWS_USER_AGENT=init\n" +
+	expectedContent1 := "export FLB_AWS_USER_AGENT=ecs-init\n" +
```
Comment (johnjameswhitman) [inline init/fluent_bit_init_process_test.go:111]: The default for this value appears to have changed in db3fd7f9659151d629e93107818c3b8dbe9ddd0f.

**Comment by PettitWesley** [comment]
@johnjameswhitman Thank you for contributing this!

Unfortunately, the team is a bit busy right now working on these: https://github.com/aws/aws-for-fluent-bit/issues/542

I'll try to get you some review as soon as we can.

**Review by PettitWesley — COMMENTED** [review]
Thank you for this! I think the code looks good. Only thing I need to do to approve and merge this is run the manual test here: https://github.com/aws/aws-for-fluent-bit/tree/mainline/integ/test_init

Then we can include it in one of the next releases.

**Comment by johnjameswhitman** [comment]
Thanks for the review @PettitWesley -- did you want me to run that test, or are you gonna do that yourself?

**Review by Claych — APPROVED** [review]

**Comment by Claych** [comment]
init integ test passed with results shown in the screenshot
image

**Pull request merged**